packages/superdough/superdough.mjs

```diff
@@ -17,11 +17,72 @@ import { loadBuffer } from './sampler.mjs';
 export const soundMap = map();
 
 export function registerSound(key, onTrigger, data = {}) {
-  soundMap.setKey(key, { onTrigger, data });
+  soundMap.setKey(key.toLowerCase(), { onTrigger, data });
+}
+
+function aliasBankMap(aliasMap) {
+  // Make all bank keys lower case for case insensitivity
+  for (const key in aliasMap) {
+    aliasMap[key.toLowerCase()] = aliasMap[key];
+  }
+
+  // Look through every sound...
+  const soundDictionary = soundMap.get();
+  for (const key in soundDictionary) {
+    // Check if the sound is part of a bank...
+    const [bank, suffix] = key.split('_');
+    if (!suffix) continue;
+
+    // Check if the bank is aliased...
+    const aliasValue = aliasMap[bank];
+    if (aliasValue) {
+      if (typeof aliasValue === 'string') {
+        // Alias a single alias
+        soundDictionary[`${aliasValue}_${suffix}`.toLowerCase()] = soundDictionary[key];
+      } else if (Array.isArray(aliasValue)) {
+        // Alias multiple aliases
+        for (const alias of aliasValue) {
+          soundDictionary[`${alias}_${suffix}`.toLowerCase()] = soundDictionary[key];
+        }
+      }
+    }
+  }
+
+  // Update the sound map!
+  // We need to destructure here to trigger the update
+  soundMap.set({ ...soundDictionary });
+}
+
+async function aliasBankPath(path) {
+  const response = await fetch(path);
+  const aliasMap = await response.json();
+  aliasBankMap(aliasMap);
+}
+
+/**
+ * Register an alias for a bank of sounds.
+ * Optionally accepts a single argument map of bank aliases.
+ * Optionally accepts a single argument string of a path to a JSON file containing bank aliases.
+ * @param {string} bank - The bank to alias
+ * @param {string} alias - The alias to use for the bank
+ */
+export async function aliasBank(...args) {
+  switch (args.length) {
+    case 1:
+      if (typeof args[0] === 'string') {
+        return aliasBankPath(args[0]);
+      } else {
+        return aliasBankMap(args[0]);
+      }
+    case 2:
+      return aliasBankMap({ [args[0]]: args[1] });
+    default:
+      throw new Error('aliasMap expects 1 or 2 arguments, received ' + args.length);
+  }
```

Comment:
if theres a better place for this function to live lemme know and i'll move it

Reply:
looks good